pyproject.toml

```
@@ -26,7 +26,8 @@ urls = {Homepage = "https://ipython.org"}
 requires-python = ">=3.8"
 dependencies = [
     "debugpy>=1.6.5",
-    "ipython>=7.23.1",
+    "ipython>=7.23.1<=8.12;python_version=='3.8'",
```

```suggestion
    "ipython>=7.23.1,<8.13;python_version=='3.8'",
```

This should fix the error in the package tests.